### DISCUSSION TITLE
How does SFTPGO handle back pressure?

### DISCUSSION THREAD

**Discussion by Smithx10** [drakkan/sftpgo#532] [Q&A]
Is there a place I can read about how we are handling back pressure?   I think this would be a good addition to the documentation, if it isn't already.  I've only git grep'd the grep searching.  If operators are going to deploy this,  knowing how much DRAM / Storage that will be required is ideal.

Thanks Again for a great piece of software.  It's been a pleasure to use and extend.

**Comment by drakkan** [comment]
Hi,

A while ago I did some tests with around 1000 concurrent connections on a low-end server with no problems, but I haven't recorded exact results.

The protocol and the authentication method matters. For example, the last time I ran this test, I noticed that using argon2id as a password hashing, if we have many simultaneous password logins we can easily run out of server memory. This happens because argon2id requires a lot of memory by design, you can adjust the memory usage by editing the configuration file. To avoid unexpected issues with the default configuration we switched to bcrypt as the default password hashing after this test.

If you want to record the results for a such test on a reference hardware, I would be glad to accept a PR and add it to the docs, thanks
